src/xrpld/overlay/detail/TrafficCount.h

```diff
@@ -175,20 +195,24 @@ class TrafficCount
             cat <= category::unknown,
             "ripple::TrafficCount::addCount : valid category input");
 
+        auto const& it = counts_.find(cat);
```

Comment:
[nit]
1) It is wrong to make it `const`, since you are modifying values later.
2) no need for a reference, because `find` returns a value.
```suggestion
        auto it = counts_.find(cat);
```

Reply:
Could you help me understand why is it wrong, I'm not modifying the iterator, I'm modifying the underlying value? From my understanding from the documentation, `find` returns an iterator. Thus in this case `const` is stating that the iterator cannot be modified, not the underlying value, no?

Reply:
Done!

Reply:
@vlntb I'm happy with your change, but confusion aside, why is it wrong to make the iterator constant?

According to the documentation `.find([key])` returns an iterator.  `auto const it = counts_.find(cat);`  in essence is stating that the iterator is constant, however it does not make any assumptions about whether the value can be modified.  In principle I'm stating there, that once you find the iterator in the map, you can do anything you want with the value, but you cannot change the iterator. (https://godbolt.org/z/rjb4d3q8P)

Reply:
You are correct differentiating between constant iterator and immutability of the contents. If the intention was to keep iterator constant but to allow modification of the data, then your change is correct. However, my preference was to keep things simple and consistent here - if you are allowing to modify underlying data, let's make the whole thing non-const.